### ISSUE TITLE
Tools to build website linked to repo

### ISSUE THREAD

**Issue by Mitzi-Laszlo** [solid/information#249]
Does anyone have any tips about how to build a website with a modern design that is linked to a GitHub repository? 

Thank you @Vinnl for putting together https://solid.github.io/information/ with GitHub pages and thank you @maxidorius for the tips on Travis-CI and https://gitlab.com/kamax-io/websites/www.kamax.io. 

@justinwb is having a look at options, tips are very welcome.

**Comment by Vinnl** [comment]
Depends on what you're missing on https://solid.github.io/information/ :) If it's the theme, there are a number of themes available through GitHub. It's using Jekyll to generate the website, so theoretically you should be able to use any [Jekyll](https://jekyllthemes.io/) [theme](http://jekyllthemes.org/), but I think GitHub doesn't allow that much configuration.

If so, it'd be relatively easy to switch to GitLab Pages, which works similarly but does allow for all that customisation. You'd probably want to use a custom domain for that then, though, since otherwise it'd be solid.gitlab.io/information, which is probably confusing.

I'd offer to help, but given that I'll be offline the coming period, that might not be quick enough :)

**Comment by maxidorius** [comment]
The theme you want to use will actually decide the tool you'll need to use. If it was created just for Solid, then it is safe to assume any tool will work. It's just a matter of porting it.

If the theme is not custom, and relies on another, then it's another approach: some themes are only available or usable with some tools. While it is always possible to "port" a theme to another tool, it can be very difficult. It all depends on the theme and its complexity.

Once you know what you can do with your theme, you'll directly know which tool you can use to build the website. And once you know that, the rest is obvious.

The other side is hosting: As long as there is a SSH or FTP access or anything similar, we can always push the files after they have been built.

Short version: let's start by getting in touch with the people who created the current theme, or to the source files of the theme.

**Comment by justinwb** [comment]
> Depends on what you're missing on https://solid.github.io/information/ :) If it's the theme, there are a number of themes available through GitHub. It's using Jekyll to generate the website, so theoretically you should be able to use any [Jekyll](https://jekyllthemes.io/) [theme](http://jekyllthemes.org/), but I think GitHub doesn't allow that much configuration.

The current plan is to use jekyll, and if we can - host it through github pages. However, because  github pages limits what extensions you can use with jekyll, we may host it elsewhere, and have some automation that rebuilds the site when changes are pulled into the master branch.

There has been some consideration for other SSGs - like Hugo - but people seem to be most familiar with jekyll and if we can get away with ease of hosting through github pages all the better.

**Comment by Vinnl** [comment]
Yeah, if you're on GitHub Pages, you'll have to make do with its restrictions. You can connect GitLab to a GitHub repository and have it build and deploy to GitLab Pages using Jekyll or a different generator - its just a CI in that regard, but including hosting.
